docs: document upgrade backup files (#2275)
edgelesssys/constellation: docs/docs/workflows/upgrade.md

For each node in your cluster, a new node has to be created and joined.
The process usually takes up to ten minutes per node.

> When applying an upgrade, backup files of the Terraform state and certain Custom Resources are created.

Reviewer: ```suggestion
When applying an upgrade, backup files of Constellation-managed Custom Resource Definitions, Custom Resources, and Terraform state are created.
```

> The Terraform state backup can be used as a reference to the possible Terraform resource migration performed within an upgrade.

Reviewer: ```suggestion
The Terraform state backup can be used to restore previous resources in case an upgrade misconfigured or erroneously deleted a resource. 
```

> The Custom Resource backup files can be used to restore Custom Resources manually (e.g. via `kubectl apply`) if the automatic migration of those resources fails.

Reviewer: ```suggestion
The Custom Resource (Definition) backup files can be used to restore Custom Resources and Definitions manually (e.g. via `kubectl apply`) if the automatic migration of those resources fails.
```

> The Terraform state backup can be used to restore previous resources in case an upgrade misconfigured or erroneously deleted a resource. 
> The Custom Resource (Definition) backup files can be used to restore Custom Resources and Definitions manually (e.g. via `kubectl apply`) if the automatic migration of those resources fails.

Reviewer 2: ```suggestion
You can use the Terraform state backup to restore previous resources in case an upgrade misconfigured or erroneously deleted a resource.
You can use the Custom Resource (Definition) backup files to restore Custom Resources and Definitions manually (e.g., via `kubectl apply`) if the automatic migration of those resources fails.
```
nit: prefer active voice (MS style guide)
